File: gc_content.py
from Bio import SeqIO

# Read FASTA file
record = SeqIO.read("gene.fna", "fasta")

# Get DNA sequence
sequence = str(record.seq).upper()

# Count nucleotides
a = sequence.count("A")
t = sequence.count("T")
g = sequence.count("G")
c = sequence.count("C")

# Calculate GC content
gc_content = (g + c) / len(sequence) * 100

# Display results
print("Gene:", record.id)
print("Sequence Length:", len(sequence))
print("Adenine (A):", a)
print("Thymine (T):", t)
print("Guanine (G):", g)
print("Cytosine (C):", c)
print("GC Content: {:.2f}%".format(gc_content))

File: gc_content.R
library(Biostrings)

# Read FASTA file
seq <- readDNAStringSet("gene.fna")

# Count G and C
gc <- letterFrequency(seq, letters = c("G", "C"))

# Calculate GC percentage
gc_percent <- sum(gc) / sum(width(seq)) * 100

# Display result
cat("GC Content:", round(gc_percent, 2), "%\n")
